aws_lambda_builders/workflows/nodejs_npm/actions.py

```diff
@@ -177,22 +175,45 @@ def __init__(self, artifacts_dir, osutils):
         :param osutils: An instance of OS Utilities for file manipulation
         """
 
-        super(NodejsNpmrcCleanUpAction, self).__init__()
+        super(NodejsCleanUpAction, self).__init__()
         self.artifacts_dir = artifacts_dir
         self.osutils = osutils
 
     def execute(self):
         """
-        Runs the action.
+        Runs the actions
 
-        :raises lambda_builders.actions.ActionFailedError: when .npmrc copying fails
+        Raises
+        ------
+        ActionFailedError
+            When .npmrc removing or package.json restoring fails
         """
 
         try:
-            npmrc_path = self.osutils.joinpath(self.artifacts_dir, ".npmrc")
-            if self.osutils.file_exists(npmrc_path):
-                LOG.debug(".npmrc cleanup in: %s", self.artifacts_dir)
-                self.osutils.remove_file(npmrc_path)
-
+            self._remove_npmrc()
+            self._restore_package_json()
         except OSError as ex:
             raise ActionFailedError(str(ex))
+
+    def _remove_npmrc(self):
+        """
+        Removes the .npmrc file
+        """
+        npmrc_path = self.osutils.joinpath(self.artifacts_dir, ".npmrc")
+
+        if self.osutils.file_exists(npmrc_path):
+            LOG.debug(".npmrc cleanup in: %s", self.artifacts_dir)
+            self.osutils.remove_file(npmrc_path)
+
+    def _restore_package_json(self):
+        """
+        Restores the original package.json from its backup
+        """
+        org_file = "package.json"
+        bak_file = org_file + ".bak"
+
+        for (root, _, files) in self.osutils.walk(self.artifacts_dir):
```

Comment:
Why is the walk necessary?

Reply:
During the building process we update the `package.json` files to point to the new packed dependencies.
We make backups of the original files.
At the end of the process we walk the dirs to revert the files to their backup so that the files we upload are the original ones. (https://github.com/aws/aws-lambda-builders/issues/187)